cmd/proxygenerator/interceptor.go

```diff
@@ -104,6 +109,30 @@ func NewPayloadVisitorInterceptor(options PayloadVisitorInterceptorOptions) (grp
 
 		err := invoker(ctx, method, req, response, cc, opts...)
 		if err != nil {
+			if options.Inbound != nil {
+				stat, ok := status.FromError(err)
+				if ok {
+					// user provided payloads can sometimes end up in the status details of
+					// gRPC errors, make sure to visit those as well
+					newStatus := status.New(stat.Code(), stat.Message())
+					var newDetails []protoadapt.MessageV1
+					for _, detail := range stat.Details() {
+						detailAny, ok := detail.(*anypb.Any)
```

Comment:
What if details contains a proto message that is not any but contains a payload?

Reply:
In theory this should be impossible looking at source, but I agree, we might as well first type assert to message v1, _then_ optionally type assert to any (falling back to just appending the v1 message if it didn't work)

Reply:
by "theory this should be impossible looking at source" do you mean the server source or go protobuf?